lib/macho/headers.rb

```diff
@@ -505,30 +505,14 @@ module Headers
     # @see MachO::FatArch
     class FatHeader < MachOStructure
       # @return [Integer] the magic number of the header (and file)
-      attr_reader :magic
+      field :magic, :uint32_net
 
       # @return [Integer] the number of fat architecture structures following the header
-      attr_reader :nfat_arch
-
-      # always big-endian
-      # @see MachOStructure::FORMAT
-      # @api private
-      FORMAT = "N2"
-
-      # @see MachOStructure::SIZEOF
-      # @api private
-      SIZEOF = 8
-
-      # @api private
-      def initialize(magic, nfat_arch)
-        super()
-        @magic = magic
-        @nfat_arch = nfat_arch
-      end
+      field :nfat_arch, :uint32_net
```

Comment:
This is a very minor thing, but instead of having a separate `:uint32_net` specifier we should try something like:

```ruby
field :nfat_arch, :uint32, :endian => :big
```

Where `:endian` is not specified by default for fields that have their endianness determined by the overarching container.

Reply:
Oh yeah, that would look nicer. I'll see what I can do.